ament_cmake_google_benchmark/ament_cmake_google_benchmark/__init__.py

```diff
@@ -53,3 +80,61 @@ def main(argv=sys.argv[1:]):
             res.returncode = 1
 
     return res.returncode
+
+
+def convert_google_benchark_to_jenkins_benchmark(
+    in_file,
+    out_file,
+    package_name=None,
+    overlay_path=None
+):
+    in_data = json.load(in_file)
+    group_name = os.path.basename(in_data['context']['executable'])
+    if package_name:
+        group_name = '%s.%s' % (package_name, group_name)
+
+    out_data = {
+        group_name: {},
+    }
+    for benchmark in in_data.get('benchmarks', []):
+        out_data[group_name][benchmark['name']] = {
+            'parameters': {
+                'iterations': {
+                    'value': benchmark['iterations'],
+                },
+            },
+            'cpu_time': {
+                'dblValue': benchmark['cpu_time'],
+                'unit': benchmark['time_unit'],
+            },
+            'real_time': {
+                'dblValue': benchmark['real_time'],
+                'unit': benchmark['time_unit'],
+            },
+        }
+
+        # Add any "additional" metrics besides cpu_time and real_time
+        out_data[group_name][benchmark['name']].update({
+            extra_name: {
+                'value': benchmark[extra_name],
+            } for extra_name in set(benchmark.keys()) - extra_metric_exclusions})
+
+    if not out_data[group_name]:
+        print(
+            'WARNING: The perfromance test results file contained no results',
```

Comment:
@cottsay nit: 
```suggestion
            'WARNING: The performance test results file contained no results',
```

Reply:
Also, should this `raise` instead and leave the caller to handle the error? I believe json serialization and deserialization can also raise.

Reply:
> ... should this `raise` instead ...?

I think it's reasonable to allow the benchmark run to generate no results, possibly due to arguments that end up skipping all of the tests. As far as I can tell, Google Benchmark seems to generate a completely empty file when this happens (which I'm now handling in 25a94a8), but I could see it being possible to generate valid JSON that doesn't contain any results we're interested in.

I'll probably need to make a follow-up PR after this one gets merged to handle more than just the "iteration" type of benchmarks. Right now, the "aggregate" types are causing parse problems. After that change, we'd specifically be converting only specific metrics, making it a realistic possibility that we'd encounter valid JSON but no convertible results, despite there not being any problems.